src/libs/SearchQueryUtils.ts

```diff
@@ -827,6 +828,32 @@ function getCurrentSearchQueryJSON() {
     return queryJSON;
 }
 
+/**
+ * Extracts the query text without the filter parts.
+ * This is used to determine if a user's core search terms have changed,
+ * ignoring any filter modifications.
+ *
+ * @param searchQuery - The complete search query string
+ * @returns The query without filters (core search terms only)
+ */
+function getQueryWithoutFilters(searchQuery: string) {
+    const parsedQuery = parseForAutocomplete(searchQuery);
+    const lastFilter = parsedQuery?.ranges.at(-1);
+
+    if (!lastFilter) {
+        return searchQuery.trim();
+    }
+
+    const queryJSON = buildSearchQueryJSON(searchQuery);
+    if (!queryJSON) {
+        return searchQuery.trim();
```

Comment:
```suggestion
        return '';
```
NAB. I think it's safer to return an empty string for this case.